tests/integration/pdo/mysql/base-class/test_query_3.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*DESCRIPTION
+The agent should record database metrics for the FETCH_CLASS variant of
+PDO::query() when PDO base class constructor is used to create connection
+object.
+*/
+
+/*SKIPIF
+<?php require(realpath (dirname ( __FILE__ )) . '/../../skipif_mysql.inc');
+*/
+
+/*INI
+newrelic.datastore_tracer.database_name_reporting.enabled = 0
+newrelic.datastore_tracer.instance_reporting.enabled = 0
+*/
+
+/*EXPECT_ERROR_EVENTS null*/
+
+/*EXPECT
+ok - create table
+ok - insert row
+ok - fetch row as object
+ok - drop table
+*/
+
+/*EXPECT_METRICS_EXIST
+Datastore/all, 4
+Datastore/allOther, 4
+Datastore/MySQL/all, 4
+Datastore/MySQL/allOther, 4
+Datastore/operation/MySQL/create, 1
+Datastore/statement/MySQL/test/create, 1
+Datastore/operation/MySQL/insert, 1
+Datastore/statement/MySQL/test/insert, 1
+Datastore/operation/MySQL/select, 1
+Datastore/statement/MySQL/test/select, 1
+Datastore/operation/MySQL/drop, 1
+Datastore/statement/MySQL/test/drop, 1
+*/
+
+require_once(realpath (dirname ( __FILE__ )) . '/../../test_query_3.inc');
+require_once(realpath (dirname ( __FILE__ )) . '/../../../../include/config.php');
+
+test_pdo_query(new PDO($PDO_MYSQL_DSN, $MYSQL_USER, $MYSQL_PASSWD), 0);
```

Comment:
Are there any concerns about interactions of test runs when a test fails?  I'm thinking of each test creating and destroying a table - is it possible a table drop doesn't occur due to a failure, leading to an unclean test environment for the next test?

Reply:
That doesn't really address my concern - but I suppose if there is a failure of this nature it will be a bit trickier to troubleshoot but should be fairly obvious the cause was the previous test left the environment in an unclean state.